numba/unicode.py

```diff
@@ -2263,6 +2266,50 @@ def _unicode_swapcase(data, length, res, maxchars):
 def unicode_swapcase(data):
     return case_operation(_ascii_swapcase, _unicode_swapcase)
 
+
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Python/bltinmodule.c#L1781-L1824    # noqa: E501
+@overload(ord)
+def ol_ord(c):
+    if isinstance(c, types.UnicodeType):
+        def impl(c):
+            lc = len(c)
+            if lc != 1:
+                # CPython does TypeError
+                raise TypeError("ord() expected a character")
+            return _get_code_point(c, 0)
+        return impl
+
+
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L2005-L2028    # noqa: E501
+@register_jitable
```

Comment:
This seemingly differs significantly from the cpython implementation referenced above. It might be worthwhile adding a comment or two about it.

Reply:
It's logically the same, just looks different because the branching etc is abstracted in a different way, I've added a comment to hopefully help.